src/basic/missing_syscall.h

```diff
@@ -75,40 +75,6 @@ static inline int memfd_create(const char *name, unsigned int flags) {
 /* ======================================================================= */
 
 #if !HAVE_GETRANDOM
-#  ifndef __NR_getrandom
-#    if defined __x86_64__
-#      define __NR_getrandom 318
-#    elif defined(__i386__)
-#      define __NR_getrandom 355
-#    elif defined(__arm__)
-#      define __NR_getrandom 384
-#   elif defined(__aarch64__)
-#      define __NR_getrandom 278
-#    elif defined(__ia64__)
-#      define __NR_getrandom 1339
-#    elif defined(__m68k__)
-#      define __NR_getrandom 352
-#    elif defined(__s390x__)
-#      define __NR_getrandom 349
-#    elif defined(__powerpc__)
-#      define __NR_getrandom 359
-#    elif defined _MIPS_SIM
-#      if _MIPS_SIM == _MIPS_SIM_ABI32
-#        define __NR_getrandom 4353
-#      endif
-#      if _MIPS_SIM == _MIPS_SIM_NABI32
-#        define __NR_getrandom 6317
-#      endif
-#      if _MIPS_SIM == _MIPS_SIM_ABI64
-#        define __NR_getrandom 5313
-#      endif
-#    elif defined(__arc__)
-#      define __NR_getrandom 278
-#    else
-#      warning "__NR_getrandom unknown for your architecture"
-#    endif
-#  endif
-
```

Comment:
humm no. that's not what we want. we want that our code makes use of what the kernel can provide independently of what it is compiled against. People tend to switch kernels frequently and we want to make us somewhat independent of that and not expect that people recompile everything whenever they change kernels.

What do you mean by "noise in kernel logs"? when we invoke a syscall that is not supported then this shouldn't really generate noise...

If there's really benefit in avoiding to repeatedly call syscalls not implemented, then this fact should be cached in userspace instead, but we should really call the syscall at least one to see if it works.

Reply:
The idea was that it would still make use of the syscalls, since __NR_getrandom would be defined for most relevant platforms, and then it would be used, provided recent kernel headers.

Reply:
...but a more suitable fix for me is perhaps to just remove the #else -> #warning block.  I'm pretty sure that would have the same effect for me.

Reply:
...hmm, no, wait, it wouldn't.  Then __NR_getrandom would be defined, even though the syscall does not exist (due to old kernel).  What the current code does is say that "I see that your kernel headers are old, but I'm pretty sure your kernel isn't, so I'll try the syscall anyway", and in our case that's not true.  The idea behind our patch is that we let the kernel headers decide if we should try the syscall.  But I can drop the patch and keep it local if you wish.

Reply:
yes, please drop this bit. in the generic case we tend to assume that kernels are more modern than much userspace, hence we want to call the syscall regardless whether libc defines its nr or not.

I am still wondering about the "noise in kernel logs"? what precisely do you see on your kernels if we keep calling the syscall even if it doesn't exist?

Note that we already cache the fact whether the syscall exists in our code, see acquire_random_bytes(). Hence we should not bother with the syscall more than once per process